### PR TITLE
More on GMS and minor refactoring

### LINKED ISSUE #602: Sign-in pages should be disabled when Play Services is not available
When sign-in is not possible, the buttons that try to sign-in should be disabled. 

This includes 

- [ ] Sign in through Achievements, Arrow page. 
- [ ] Sign in from Settings.

### LINKED ISSUE #614: G+ Button not updated after clicking
Due to change from ListView to RecyclerView the g+ button that was clicked is not updated and continues to show an idling animation.

### PR DESCRIPTION
**Pull request by friedger**
This PR fixes #602 (hiding games menu and settings button for no-gms devices)

This PR fixes #614 (G+ button in news fragment)

This PR removes unused code.

### CODE REVIEW

File: app/src/main/java/org/gdg/frisbee/android/arrow/Organizer.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package org.gdg.frisbee.android.arrow;
 
-import com.google.android.gms.plus.model.people.Person;
+
+import org.gdg.frisbee.android.api.model.plus.Person;
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/arrow/Organizer.java]: Wow, I cannot define how I am happy to see this. :smile:

File: app/src/main/java/org/gdg/frisbee/android/cache/ModelCache.java [hunk 2]
```diff
@@ -219,14 +219,6 @@ public boolean isMemoryCacheEnabled() {
         return null != mMemoryCache;
     }
 
-    public CacheItem put(final String url, final Object obj) {
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/cache/ModelCache.java]: Wohooo

File: app/src/main/java/org/gdg/frisbee/android/fragment/SettingsFragment.java [hunk 3]
```diff
@@ -147,28 +150,34 @@ public void onNotFound(String key) {
...
-                        PrefUtils.setLoggedOut(getActivity());
-                        createConnectedGoogleApiClient();
+                        return true;
                     }
-                    return true;
-                }
-            });
+                });
+            } else {
+                PreferenceScreen root = (PreferenceScreen) findPreference(PrefUtils.SETTINGS_ROOT);
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/fragment/SettingsFragment.java]: :+1: I was thinking to disable it and set the description to something like this: "Play Services not available"

File: app/src/main/java/org/gdg/frisbee/android/chapter/NewsAdapter.java [hunk 4]
```diff
@@ -196,9 +200,9 @@ public void onBindViewHolder(ViewHolder holder, int i) {
 
         holder.url = activity.getUrl();
 
-        if (mPlusClient != null && mPlusClient.isConnected()) {
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/chapter/NewsAdapter.java]: This is really similar to `updatePlusOne` method.

File: app/src/main/java/org/gdg/frisbee/android/chapter/NewsFragment.java [hunk 5]
```diff
@@ -55,8 +56,13 @@ public static NewsFragment newInstance(String plusId) {
     public void onResume() {
         super.onResume();
 
-        for (int i = 0; i <= getListView().getChildCount(); i++) {
-            mAdapter.updatePlusOne(getListView().getChildAt(i));
+        RecyclerView recyclerView = getListView();
+        for (int i = 0; i <= recyclerView.getChildCount(); i++) {
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/chapter/NewsFragment.java]: Should we cache the size inside the for loop? 

`for (int i = 0, size = recyclerView.getChildCount(); i <= size; i++)`

**Comment by friedger** [comment]
Comments addressed

**Comment by tasomaniac** [comment]
LGTM

**Pull request merged**